pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -222,9 +222,32 @@ func (pc *PodCache) getPodByIP(addr string) *v1.Pod {
 	if !exists {
 		return nil
 	}
-	item, exists, err := pc.informer.GetStore().GetByKey(key)
-	if !exists || err != nil {
-		return nil
+	return pc.getPodByKey(key)
+}
+
+// getPodByKey returns the pod or nil if pod not found or an error occurred
+func (pc *PodCache) getPodByKey(key string) *v1.Pod {
+	item, _, _ := pc.informer.GetStore().GetByKey(key)
+	if item != nil {
+		return item.(*v1.Pod)
 	}
-	return item.(*v1.Pod)
+	return nil
+}
+
+// getPodByKey returns the pod of the proxy
+func (pc *PodCache) getPodByProxy(proxy *model.Proxy) *v1.Pod {
+	var pod *v1.Pod
+	key := proxyPodKey(proxy)
+	if key != "" {
+		pod = pc.getPodByKey(key)
+		if pod != nil {
+			return pod
+		}
+	}
+
+	// only need to fetch the corresponding pod through the first IP, although there are multiple IP scenarios,
+	// because multiple ips belong to the same pod
+	proxyIP := proxy.IPAddresses[0]
```

Comment:
This may panic if we have no IPs? Not sure if that is possible at this point though

Reply:
It is not possible, such usage is widely in this repo